### PR TITLE
Fixes Issue #803

### PR DESCRIPTION
**Pull request by aravindMahadevan**
Modify the _decode_asr method to support decoding user defined token in Whisper based models. Addresses issue in #803

### CODE REVIEW

File: src/tokenizers.js
Comment (xenova) [inline src/tokenizers.js:3613]:
```suggestion
        const timestamp_end = this.model.convert_tokens_to_ids(["<|30.00|>"])[0];
```
to get the token id of a single element

**Review by xenova — CHANGES_REQUESTED** [review]
Looks good! Just one thing:

File: src/tokenizers.js
Comment (aravindMahadevan) [reply src/tokenizers.js]: Commited suggestion, thanks for catching!

**Comment by aravindMahadevan** [comment]
@xenova committed the fix, let me know if there is anything else needed!

**Review by avinashr175 — APPROVED** [review]
lgtm!

**Comment by xenova** [comment]
Thanks! Will merge after tests pass. By the way, do you have an example of a whisper model which such tokens? Might be good to add a test.

**Comment by aravindMahadevan** [comment]
Hi @xenova, there are two tests failing due to a few test models not having the <30.00> token. Here's one such example:  https://huggingface.co/Xenova/whisper-small/resolve/output_attentions/tokenizer.json. 

Do you have any suggestions on how to overcome this issue? This token does exist in all of the base whisper models.

**Comment by xenova** [comment]
We could probably use the time precision (0.02) to calculate the offset: 30/0.02 + 1 = 1501 tokens (50364 -> 51864). Another fix is to simply update the tokenizer.json.

**Comment by aravindMahadevan** [comment]
Hi @xenova , that was a good suggestion and I have updated the logic to the following: `const timestamp_end = timestamp_begin + total_timestamp_tokens`
`const total_timestamp_tokens = (30.00 - 0.00) / 0.02`

This logic should work on both English only Whisper and multilingual Whisper variants. The beginning timestamp of 0.00 is equal to token id 50363 and 50364 in the English only Whisper variants and Multilingual Whisper variants respectively. Similarly, the final timestamp token id of 30.00 is equal to 51863 and 51864 in the English only Whisper variants and Multilingual Whisper variants respectively as well. In both cases, the final timestamp token id occurs at an offset of 1500 which is what total_timestamp_tokens evaluates to. 

I do not have a model with added tokens that I can share publicly. I did find this model https://huggingface.co/oza75/whisper-bambara-asr-001 which has added tokens after the final timestamp but it's a special token.

**Comment by aravindMahadevan** [comment]
Hey @xenova, are there any more changes that should be made to this PR?

**Comment by radai-jamescheng** [comment]
@xenova Hi! I believe all test cases are passing, is there anything else needed before we can merge this change? Would you be able to take one last look? Thanks!

**Review by xenova — APPROVED** [review]
Apologies for the delay... This PR got lost in my notifs...

Thanks again! 🙏

File: src/tokenizers.js
Comment (xenova) [inline src/tokenizers.js:3617]:
```suggestion
        const timestamp_begin = this.timestamp_begin;
        // Whisper timestamp tokens start from 0.00 and go to timestamp 30.00 in 0.02 increments. 
        // We can calculate the last time stamp token as timestamp_begin plus the number of tokens 
        // tokens from 0.00 to 30.00 which is 1500. 
        const total_timestamp_tokens = 1500; // (30.00 - 0.00) / 0.02
        const timestamp_end = timestamp_begin + total_timestamp_tokens;
```

**Pull request merged**
